src/components/pages/templates-manage/TemplatesManage.js

```diff
@@ -8,44 +8,48 @@ import { injectIntl } from 'react-intl';
 import { Form, Input } from '../../form/Form';
 import Select from 'react-select';
 import Loader from '../../ui/Loader';
+import { getSelectorValueFromArray } from '../../../helpers/filters';
 
 class TemplatesManage extends React.Component {
   constructor (props) {
     super(props);
-    this.state = {
-      selectedArea: this.props.selectedArea || null,
-      defaultLanguage: this.props.selectedArea || null
-    }
-    this.form = {
-      title: props.template? props.template.name.en : null
-    };
+    this.state = {};
+  }
+
+  componentDidMount() {
+    if (this.props.template) this.setPropsToForm(this.props);
   }
 
   componentWillReceiveProps(nextProps) {
-    this.form = {
-        ...this.form,
-        title: nextProps.template !== null ? nextProps.template.name.en : ''
-    };
+    if (nextProps.template !== this.props.template) this.setPropsToForm(nextProps);
+  }
+
+  setPropsToForm = (props) => {
+    this.setState({
+      title: props.template.name.en || null,
```

Comment:
Yes, this is place holder for now as MANY :) things depend on this so will be in next PR. Super complicated.